app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Users
+  class RegistrationsController < Devise::RegistrationsController
+    # before_action :configure_sign_up_params, only: [:create]
+    # before_action :configure_account_update_params, only: [:update]
+
+    # GET /resource/sign_up
+    # def new
+    #   super
+    # end
+
+    # POST /resource
+    # def create
+    #   super
+    # end
+
+    # GET /resource/edit
+    # def edit
+    #   super
+    # end
+
+    # PUT /resource
+    # def update
+    #   super
+    # end
+
+    # DELETE /resource
+    # def destroy
+    #   super
+    # end
+
+    # GET /resource/cancel
+    # Forces the session data which is usually expired after sign
+    # in to be expired now. This is useful if the user wants to
+    # cancel oauth signing in/up in the middle of the process,
+    # removing all OAuth session data.
+    # def cancel
+    #   super
+    # end
+
+    # protected
+
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_sign_up_params
+    #   devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
+    # end
+
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_account_update_params
+    #   devise_parameter_sanitizer.permit(:account_update, keys: [:attribute])
+    # end
+
+    # The path used after sign up.
+    def after_sign_up_path_for(resource)
+      UserMailer.welcome_email(resource).deliver_now
```

Comment:
I don't like this here, because this method name means that only after_sign_up_PATH should be there. Also, it could be a situation when user is persisted, but he is not active for authentication and then the email won't be sent.

Reply:
you can check if the resource is persisted before email sending